runtime/filetype.vim

```diff
@@ -52,7 +52,10 @@ func s:StarSetf(ft)
 endfunc
 
 " Vim help file
-au BufNewFile,BufRead $VIMRUNTIME/doc/*.txt	setf help
+au BufNewFile,BufRead */doc/*.txt
+	\  if getline('$') =~ 'vim:.*\<\(ft\|filetype\)=help\>'
```

Comment:
The `vim` should be either preceded by whitespace or at the start of the line. This pattern doesn't check for that.

Reply:
Yeah, it also doesn't parse some of the other modeline formats. I started with the simple regex below and modified it to work with all the modelines in runtime/doc.

https://github.com/vim/vim/blob/4422de6316b544c282e6c74afd3df3ee3a4b1cfd/runtime/filetype.vim#L3321-L3325

In the description of #16777 I used some more complicated regexes to parse modelines more completely, but those are a hard-to-read mess. Do you want

1. more correct modeline parsing using something like the regexes in #16777?
2. a new vimscript function to parse modelines using the same code that vim does itself?
3. the current solution?
4. something else?

Reply:
I just want to avoid false positives :)

Reply:
Fair. I wish vim used a different extension like .vimhelp or something to make this easier. Or even .vim.txt if ending in .txt is useful.

Reply:
Would adding `\<` at the beginning of the regex help? It's simple and doesn't take much effort, like adding a new modeline parsing function would. But I'm not sure how much it'll cut down on false positives, e.g.:

```
$ grep -i 'vim$' /usr/share/dict/american-english 
vim
```

Reply:
I was thinking more about the case where some text quotes a modeline in double quotes or backticks, where `\<` will not help. But adding `\(^\|\s\)` should work.

Reply:
I've opened a PR #16845

Reply:
Gotcha. No idea how common that is, but makes sense.